### PR TITLE
Add node.js examples

### PR DESCRIPTION
**Pull request by zsiciarz**
I'm adding examples of node.js bindings using [ffi](https://www.npmjs.com/package/ffi). Don't merge yet - this is a work in progress and needs Travis integration and at least a few words of documentation on the Jekyll site. I'll focus on that next and will update this PR later.

### CODE REVIEW

**Comment by shepmaster** [comment]
Looking good so far! You are the first to mess with my Makefile setup besides myself; how easy was it to pick up and understand? I know I should add some docs to them, do you have any suggestions on what would be highest value?

**Comment by zsiciarz** [comment]
To be frank, I copy-pasted the makefiles from Python examples and updated accordingly hoping I don't mess up anything ;-) Turned out OK, but I think it would be valuable to have at least some explanation of the directory structure, various includes and the flow of control. 

BTW, I've fixed Travis build so only the docs are left.

File: .travis.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ addons:
 before_install:
 - bundle install
 - cd examples && bundle install ; cd ..
+- npm install ffi ref ref-array
```
Comment (shepmaster) [inline .travis.yml]: Should we have an `examples/package.json` instead?

**Comment by shepmaster** [comment]
FYI, I just pushed a new page about tuples. You can either decide to include it, or I can add it when this PR gets merged - your choice!

File: .travis.yml (hunk 1 above)
Comment (zsiciarz) [inline .travis.yml]: You're right! Just added it.

**Comment by zsiciarz** [comment]
I've added the tuples example, so I think this PR is ready to merge.

**Comment by shepmaster** [comment]
This looks great! Thanks so much!

**Pull request merged**